examples/deep_learning/distributed_flax_imagenet.py

```diff
@@ -0,0 +1,839 @@
+# Copyright 2022 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+SPMD ResNet example with Flax and JAXopt.
+====================================
```

Comment:
I would add a short description, such as the one in the pull-request description. Briefly explaining concepts such as SPMD and pjit would be useful too. Also maybe add some instructions on how to run it?

Reply:
Done! Perhaps in the future we should add individual markdown instructions for the "heavier" examples to cover things like setting up the GCP VM instances?